doc/whats-new.rst

```diff
@@ -9,6 +9,19 @@ What's New
     import xray
     np.random.seed(123456)
 
+v0.6.1 (unreleased)
+-------------------
+
+This release includes several bug fixes and enhancements.
+
+Enhancements
+~~~~~~~~~~~~
+
+Bug fixes
+~~~~~~~~~
+
+- Enforce that the ``DataArray.name`` attribute is a string or a NoneType (:issue:`533`). This guarantees that the ``DataArray`` can be properly written to a netCDF.
```

Comment:
Use `None` instead of `NoneType`

I would also mention Dataset variable names, e.g., "Enforce that the `DataArray.name` attribute and variable names in Datasets are a string or None."

Also, probably better to say "This guarantees that all datasets can be properly written to a netCDF file" -- because you can't directly write a DataArray to disk, anyways.